test/metabase/upload_test.clj

```diff
@@ -296,12 +299,83 @@
                     alternating (map even? (range (count bool-column)))]
                 (is (= alternating bool-column))))))))))
 
+(deftest load-from-csv-length-test
```

Comment:
I'd feel a lot safer if we test e2e this, the main reason is the `metabase.api.card/upload-csv!` is doing too many things and there are code paths we haven't tested there.

In this particular example, we are testing `unique-table-name` but from the call site, we do things a bit more complicated there.
https://github.com/metabase/metabase/blob/717a018dbc550032d55ed3fa968327efee511b7a/src/metabase/api/card.clj#L1007-L1008

Reply:
An e2e would ensure `unique-table-name` is being called on the table name. I think the confusion is this test just tests `unique-table-name`, but doesn't check that it's called. So you might as well change this to a unit test on `unique-table-name` without the `upload-load-from-csv` stuff.

Reply:
I decided to leave this how it is instead of changing it to a simple unit test on `unique-table-name` because this is also testing another property I didn't consider: that the database can handle the acceptable length of the column name given by `driver/table-name-length-limit`.